common/metadata.go

```diff
@@ -0,0 +1,235 @@
+package common
+
+type UpsertMetadataAction string
+
+const (
+	// UpsertMetadataActionCreate indicates that the object/field was created.
+	UpsertMetadataActionCreate UpsertMetadataAction = "create"
+	// UpsertMetadataActionUpdate indicates that the object/field was updated.
+	UpsertMetadataActionUpdate UpsertMetadataAction = "update"
+	// UpsertMetadataActionNone indicates that the object/field was not changed.
+	UpsertMetadataActionNone UpsertMetadataAction = "none"
+)
+
+func (a UpsertMetadataAction) IsValid() bool {
+	switch a {
+	case UpsertMetadataActionCreate,
+		UpsertMetadataActionUpdate,
+		UpsertMetadataActionNone:
+		return true
+	default:
+		return false
+	}
+}
+
+type AssociationCardinality string
+
+const (
+	associationCardinalityNotSet    AssociationCardinality = ""
+	AssociationCardinalityManyToOne AssociationCardinality = "many-to-one"
+	AssociationCardinalityOneToOne  AssociationCardinality = "one-to-one"
+)
+
+func (ac AssociationCardinality) IsValid() bool {
+	switch ac {
+	case associationCardinalityNotSet,
+		AssociationCardinalityManyToOne,
```

Comment:
can you add some comments to explain these? Especially the directionality of ManyToOne